controller/app.controller.js

```diff
@@ -20,13 +20,47 @@ const wrongPath = (req, res, next) => {
 };
 
 const getArticleById = (req, res, next) => {
-    selectArticle(req.params.id)
+    selectArticleById(req.params.id)
         .then((data) => {
-            res.status(200).send({ rows: data });
+            res.status(200).send({ article: data[0] });
+            })
+        .catch((err) => {
+            next(err);
+        });
+}
+
+const getAllArticles = (req, res, next) => {
+    selectArticles()
+        .then((data) => {
+            res.status(200).send({ articles: data });
         })
         .catch((err) => {
             next(err);
         });
 };
 
-module.exports = { getAllTopics, getApis, wrongPath, getArticleById };
+const getComments = (req, res, next) => {
+    selectAllCommentsForArticle(req.params.article_id)
+    .then((data) => {
+        if (data.length === 0){
```

Comment:
We need to check why we are returning an empty array - is it due to an invalid ID or because there are just no comments yet

As per my previous comments regarding this, In order to check whether the ID is valid we can call our earlier model where we get the article by it's ID. This model should already throw an error if the ID does not exist (assuming you added the error handling in the model and not the controller)

If we pass this and continue to get an empty array in our .then, we know the issue isn't the ID but instead that there is no comments. Therefore we are just returning the empty array.